application/models/QuestionGroup.php

```diff
@@ -163,14 +163,16 @@ public function insertRecords($data)
      * Deletes a question group and all its dependencies.
      * Returns affected rows of question group table (should be 1 or null)
      * @param integer $groupId
-     * @param integer $surveyId
+     * @param integer|null $surveyId deprecated
      * @return int|null
      */
-    public static function deleteWithDependency($groupId, $surveyId)
+    public static function deleteWithDependency($groupId, $surveyId = null)
     {
+        $QuestionGroup ::self::model()->findByPk($groupId);
```

Comment:
What if the QueestionGroup is not found? Shouldn't we have a small check after query?

Reply:
Throw 500 error maybe ?

Reply:
Thorw 500 (but must be a 404 : i report the issue)